src/Npgsql/NpgsqlTracingOptions.cs

```diff
@@ -1,9 +1,44 @@
+using System;
+using System.Diagnostics;
+
 namespace Npgsql;
 
 /// <summary>
 /// Options to configure Npgsql's support for OpenTelemetry tracing.
-/// Currently no options are available.
 /// </summary>
 public class NpgsqlTracingOptions
 {
-}
\ No newline at end of file
+    internal static NpgsqlTracingOptions? Current { get; set; }
+
+    /// <summary>
+    /// Gets or sets a filter function that determines whether or not to
+    /// collect telemetry on a per <see cref="NpgsqlCommand"/> basis.
+    /// </summary>
+    public Func<NpgsqlCommand, bool>? FilterNpgsqlCommand { get; set; }
+
+    /// <summary>
+    /// Gets or sets an action to enrich an <see cref="Activity"/> with <see cref="NpgsqlCommand"/>.
+    /// </summary>
+    public Action<Activity, NpgsqlCommand>? EnrichWithNpgsqlCommand { get; set; }
+
+    /// <summary>
+    /// Gets or sets a function that provides a span's name on a per <see cref="NpgsqlCommand"/> basis.
+    /// </summary>
+    public Func<NpgsqlCommand, string?>? ProvideSpanNameForNpgsqlCommand { get; set; }
```

Comment:
Unless `NpgsqlCommand` also exposes a property where I could store a custom span name (in which case this function wouldn't be needed in the first place) at the moment of command creation, this isn't going to be particularly helpful in my opinion. Yes, deriving the span name by using the first N characters of the command text would be trivial, but what if I want to use some label - `Select customers with payments due`, `Verify user account`, etc.? I'd have to maintain some global dictionary for mapping command texts to these labels, which would be extremely annoying.

Reply:
Not really, you can also use `Baggage.SetBaggage`/`Baggage.GetBaggage` or even a custom `AsyncLocal` to pass required info. I'm personally a bit hesitant to add a new property because it's not going to be visible if you're using ADO.NET interfaces +`NpgsqlCommand` is probably the most allocated object we have and another property there will just make it even more heavy.

Reply:
If the callback isn't executed completely out of band with respect to the created activity, then that could work.

Reply:
> If the callback isn't executed completely out of band with respect to the created activity, then that could work.

Yep, it's called just before we start doing any possible async actions (the same is true for multiplexing) so it definitely should work.

https://github.com/npgsql/npgsql/blob/43a47b95a858c121a3b1918870607206970170a4/src/Npgsql/NpgsqlCommand.cs#L1508-L1511